src/main/java/com/royalrangers/controller/UserController.java

```diff
@@ -73,63 +104,52 @@ public ResponseResult getUsersToApprove(){
-        return ResponseBuilder.success("Users successfully approved.");
+    @PostMapping("/approve/registration")
+    @PreAuthorize("hasAnyRole('ADMIN', 'SUPER_ADMIN')")
+    @ApiOperation(value = "Approve users after registration")
+    public ResponseResult approveUsers(@RequestParam Long id) {
+        try {
+            userService.approveUser(id);
+            return ResponseBuilder.success("User successfully approved.");
+        } catch (UserRepositoryException e) {
+            return ResponseBuilder.fail(e.getMessage());
+        }
     }
 
-    @PostMapping("/reject")
-    @PreAuthorize("hasRole('ADMIN')")
+    @PostMapping("/reject/registration")
+    @PreAuthorize("hasAnyRole('ADMIN', 'SUPER_ADMIN')")
     @ApiOperation(value = "Reject user after registration (for platoon admin)")
-    public ResponseResult rejectUser(@RequestBody IdsDto param) {
-        List<Long> ids = param.getIds();
-        userService.rejectUsers(ids);
-        return ResponseBuilder.success("Users successfully rejected.");
-    }
-
-    @PostMapping("/reject/super")
-    @PreAuthorize("hasRole('SUPER_ADMIN')")
-    @ApiOperation(value = "Reject users after registration (for super admin)")
-    public ResponseResult rejectUsers() {
-        userService.superRejectUsers();
-        return ResponseBuilder.success("Users successfully rejected.");
+    public ResponseResult rejectUser(@RequestParam Long id) {
+        try {
+            userService.rejectUser(id);
+            return ResponseBuilder.success("User successfully rejected.");
+        } catch (UserRepositoryException e) {
+            return ResponseBuilder.fail(e.getMessage());
+        }
     }
 
     @PutMapping("/update/temp")
-    @ApiOperation(value = "Update user data (for users)")
+    @ApiOperation(value = "Update user data (for current user)")
     public ResponseResult updateTempUser(@RequestBody UserUpdateDto update) {
-
-        String email = userService.getAuthenticatedUserEmail();
-
         userService.updateTempUser(update);
-        log.info("Update temp_user " + email);
+        log.info("Update temp_user " + userService.getAuthenticatedUser().getEmail());
 
-        return ResponseBuilder.success(String.format("User %s successfully updated", email));
+        return ResponseBuilder.success(String.format("User %s successfully updated, waiting for approve this update by admin", userService.getAuthenticatedUser().getEmail()));
     }
 
     @PutMapping("/update")
     @PreAuthorize("hasRole('ADMIN')")
-    @ApiOperation(value = "Update user data (for admin)")
-    public ResponseResult updateAuthorizedUser(@RequestBody UserUpdateDto update) {
-
-        String email = userService.getAuthenticatedUserEmail();
-
-        userService.updateUser(update);
-        log.info("Update user " + email);
+    @ApiOperation(value = "Confirm to update user data from temp_user data(for admin)")
+    public ResponseResult updateUser(@RequestParam Long id, @RequestBody UserUpdateDto update) {
```

Comment:
We need to decide whether we use **@RequestParam** or **@PathVariable**. API should we written in one style.